### DISCUSSION TITLE
Density estimators for violin plots for data in a known range

### DISCUSSION THREAD

**Discussion by kamil-wojcicki** [mwaskom/seaborn#2961] [Q&A] [answered]
If we know the range of our data, say that it is from 0 to 1 only, is there a way to use a density estimator with violin plots that will take that into account? The default KDE places symmetric kernels on each data point, and so the violins can extend well beyond the known limits. 

There are visual workarounds, e.g., using cut=0, but there the areas under the violin can be cut by substantially, and when using the area scaling, the areas of the resulting violins can differ a lot (rather than being equal).

**Comment by mwaskom** [comment, marked as answer]
No, sorry, the only option is a gaussian kernel.

**Reply by aleksejalex** [reply]
@mwaskom   ..is there a reason why not to add implementation for used-defined kernel (via lambda-function for example)?

**Reply by mwaskom** [reply]
You'd have to ask scipy.

**Reply by aleksejalex** [reply]
Aha, ok. It looks like I should implement it by myself, it would be easier and much faster solution :)
